src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsProjectAdapter.cs

```diff
@@ -406,8 +410,8 @@ private async Task<string> GetTargetFrameworkStringAsync()
             // this check. The values can be passed as null here.
             var frameworkStrings = MSBuildProjectFrameworkUtility.GetProjectFrameworkStrings(
                 projectFilePath: projectPath,
-                targetFrameworks: null,
-                targetFramework: null,
+                targetFrameworks: targetFrameworks,
+                targetFramework: targetFramework,
```

Comment:
This is modified to provide targetFramework support for C++. In addition, I found that NuGet.CommandLine does this, so I copied its implementation.

Code Here:
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.CommandLine/Common/MSBuildProjectSystem.cs#L63

Reply:
This function requires NuGet.BuildTasks configuration to use. 


Here are the changes I submitted in NuGet.BuildTasks:

https://github.com/dotnet/NuGet.BuildTasks/pull/67/commits/f1da7d0fdab637d9cf53a3caa18c2c7d74323441

Reply:
targetFramework matching order

  (vs2019)
native v14.2     
       |
  (vs2017)
native v14.1
      |
  (vs2015)              (vs2013)         (vs2012)            (vs2010)      (vs2008)
native v14.0      native v12.0     native v11.0    native v10.0    native v9.0
       ╲                      ╲                        |                    ╱                  ╱
                                                   native v0.0

Reply:
For example, using targetFramework, we can configure different static libraries for different toolsets of C++ projects.

```xml
    <dependencies>
      <group targetFramework="native14.0">
        <dependency id="Detours.Static.v140" version="4.0.1" />
      </group>
      <group targetFramework="native14.1">
        <dependency id="Detours.Static.v141" version="4.0.1" />
      </group>
      <group targetFramework="native14.2">
        <<dependency id="Detours.Static.v142" version="4.0.1" />
      </group>
    </dependencies>
```

Reply:
In the project file (vcxproj), we can also do this:

```xml
  <ItemGroup>
    <PackageReference Include="Detours.Static.v141" Condition=" '$(TargetFramework)' == 'native14.1' ">
      <Version>4.0.1</Version>
    </PackageReference>
  </ItemGroup>
```xml